modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCppCodegen.java

```diff
@@ -341,6 +349,28 @@ public void preprocessOpenAPI(OpenAPI openAPI) {
         if (!scheme.isEmpty()) {
             this.additionalProperties.put("scheme", scheme);
         }
+        if (!serverList.isEmpty()) {
+            for (Server server : serverList) {
+                CodegenServer s = new CodegenServer();
+                s.description = server.getDescription();
+                s.url = server.getUrl();
+                s.variables = new ArrayList<CodegenServerVariable>();
+                ServerVariables serverVars = server.getVariables();
+                if(serverVars != null){
+                serverVars.forEach((key,value) -> {
+                    CodegenServerVariable codegenServerVar= new CodegenServerVariable();
+                    ServerVariable ServerVar = value;
+                    codegenServerVar.name = key;
+                    codegenServerVar.description = ServerVar.getDescription();
+                    codegenServerVar.defaultValue = ServerVar.getDefault();
+                    codegenServerVar.enumValues = ServerVar.getEnum();
+                    s.variables.add(codegenServerVar);
+                    });
+                }
+                CodegenServerList.add(s);
+            }
+            this.vendorExtensions.put("x-codegen-globalServerList", CodegenServerList);
```

Comment:
Hope you don't mind. I'll rename `x-codegen-globalServerList` to `x-cpp-global-server-list` later with another PR.

Reply:
Sure thats okay. Makes more sense to stick to the kebab case.

Reply:
Filed https://github.com/OpenAPITools/openapi-generator/pull/8236 (which also removed some outdated files as part of the PR)